### PR TITLE
Website and Documentation

### PR DESCRIPTION
**Pull request by dantleech**
http://phpactor.github.io/phpactor

### CODE REVIEW

File: doc/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+---
+currentMenu: about
+---
+About Phpactor
+==============
+
+Phpactor is a intelligent code-completion and refactoring tool for PHP.
```
Comment (greg0ire) [inline doc/README.md]: an

File: doc/README.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+- **VIM plugin**: Lightweight VIM plugin.
+- **CLI interface**: Most/some functionality is exposed through CLI commands.
+
+Phpactor can be used in a limited way as a **standalone application**, but its full
+power can only be utilized through its RPC protocol. It ships with support for VIM, but it
+would not involve a huge amount of effort to support other editors such as
+[GNU Emacs](https://www.gnu.org/s/emacs/).
+
+Phpactor is a bleeding-edge product, it is not perfect, it probably never will
+be. It has been driven forward by immediate need. Comprimises have been made
```
Comment (greg0ire) [inline doc/README.md]: Compromises

File: doc/README.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+and it's a work in progress. But, I have been using it extensively in my daily
+work and havn't felt especially compelled to add new features for some months.
```
Comment (greg0ire) [inline doc/README.md]: haven't ?

File: doc/README.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+Your mileage my vary, pull requests are open!
```
Comment (greg0ire) [inline doc/README.md]: may

File: doc/history.md [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+  what PHP is good at. We locate files quickly by depending on
+  the [Composer](https://getcomposer.org) autoloader. Bad because
+  it cannot discover non-autoloadable classes or (at time of writing) provide
+  completion for functions in the wild. Phpactor does not work well without Composer.
+- **Phpactor is also a refactoring tool**: Phpactor provides [refactorings](refactorings.md) and
+  procedures to help you code faster and more accurately (such as complete
+  constructor, generate method, implement contracts etc).
+- **Phpactor ships with a VIM plugin**: Phpactor ships with a [VIM plugin](vim-plugin.md).
+
+... and competition for the eco-system right? "Competing" products allow more
```
Comment (greg0ire) [inline doc/history.md]: …

File: doc/navigation.md [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+
+Navigate / list all references to a given class.
+
+- **Command**: `$ phpactor references:class path/to/Class.php`
+- **VIM Context Menu**: _Class context menu > Find references_.
+- **VIM Command**: `call phpactor#FindReferences()`
+
+### Description
+
+Keep track of where a class is being used or perform an intial survey before
```
Comment (greg0ire) [inline doc/navigation.md]: initial

File: doc/navigation.md [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+deciding to rename a class.
+
+The VIM plugin will load the class references into a quick fix list which you can navigate through (see `:help quickfix`).
+
+The CLI command will list the references and show a highlighted line where the references was found.
```
Comment (greg0ire) [inline doc/navigation.md]: were

File: doc/navigation.md [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+- **Command**: `$ phpactor references:member path/to/Class.php memberName`
+- **VIM Context Menu**: _Member context menu > Find references_.
+- **VIM Command**: `call phpactor#FindReferences()`
+
+### Description
+
+Scan for all references to a class member in the project.
+
+This functionality is very similar to [Class References](#class-references)
+with the exception that it is possible not all members will be found as PHPHP
```
Comment (greg0ire) [inline doc/navigation.md]: - possible that?
- PHP

File: doc/refactorings.md [hunk 9]
```diff
@@ -0,0 +1,880 @@
...
+    public function goodbye(Hello $hello)
+    {
+    }
+}
+```
+
+Implement Contracts
+-------------------
+
+Add any not implemented methods from interfaces or abstract classes.
```
Comment (greg0ire) [inline doc/refactorings.md]: not => non ?

File: doc/refactorings.md [hunk 10]
```diff
@@ -0,0 +1,880 @@
...
+- **Command**: _RPC only_
+- **VIM plugin**: _Variable context menu > Rename_.
+- **VIM function**: `call phpactor#ContextMenu()`
+
+### Motivation
+
+Having meaningful and descriptive variable names makes the intention of
+code clearer and therefore easier to maintain. Renaming variables is a frequent
+refactoring, but doing this with a simple search and replace can often have unintended
+consquences (e.g. renaming the variable `$class` also changes the `class` keyword).
```
Comment (greg0ire) [inline doc/refactorings.md]: consequences

File: doc/refactorings.md [hunk 11]
```diff
@@ -0,0 +1,880 @@
...
+Rename a class member.
+
+- **Command**: `$ phpactor references:member path/to/Class.php memberName --type="method" --replace="newMemberName"` (FQN accepted)
+- **VIM plugin**: _Member context menu > Replace references_.
+- **VIM function**: `call phpactor#ContextMenu()`
+
+### Motivation
+
+Having an API which is expressive of the intent of the class is important, and
+and contribute to making your code more consistent and maintaintable.
```
Comment (greg0ire) [inline doc/refactorings.md]: - double and
- contributes

File: doc/refactorings.md [hunk 12]
```diff
@@ -0,0 +1,880 @@
...
+you may end up replacing many things you did not mean to replace (e.g. imagine renaming the method `name()`).
+
+This refactoring will:
+
+1. Scan for files in your project which contain the member name text.
+2. Parse all of the candidate files.
+3. Identify the members, and try and identify the containing class.
+4. Replace only the members which certainly belong to the target class.
+
+Due to the losely typed nature of PHP this refactoring may not find all of the member accesses for the given class. Run
```
Comment (greg0ire) [inline doc/refactorings.md]: loosely?

File: doc/refactorings.md [hunk 13]
```diff
@@ -0,0 +1,880 @@
...
+your tests before and after applying this refactoring.
+
+<div class="alert alert-info">
+<b>Hint</b>: Use the CLI command to list all of the <b>risky</b> references. Risky references are those member accesses which match
+the query but whose containg classes could not be resolved.
```
Comment (greg0ire) [inline doc/refactorings.md]: containing

File: doc/standalone.md [hunk 14]
```diff
@@ -0,0 +1,75 @@
+---
+currentMenu: standalone
+---
+Standalone
+==========
+
+Phpactor can be used as (and was designed be) a standalone CLI application,
```
Comment (greg0ire) [inline doc/standalone.md]: to be

File: doc/standalone.md [hunk 15]
```diff
@@ -0,0 +1,75 @@
...
+and this will act as a compliment if it is being used as a backend for a text
```
Comment (greg0ire) [inline doc/standalone.md]: complement?

File: doc/template/README.md [hunk 16]
```diff
@@ -0,0 +1,71 @@
...
+To set the current menu item (i.e. highlighted menu item), set the `currentMenu`
+key in the Markdown files:
+
+```markdown
+---
+currentMenu: home
+---
+
+# Welcome
+```
```
Comment (greg0ire) [inline doc/template/README.md]: might be ignored?

File: doc/vim-plugin.md [hunk 17]
```diff
@@ -0,0 +1,137 @@
...
+map <Leader>tt :call phpactor#Transform()<CR>
+
+" Generate a new class (replacing the current file)
+map <Leader>cc :call phpactor#ClassNew()<CR>
+```
+
+See the [Refactorings](refactorings.md) chapter for more functions you can map
+shortcuts to.
+
+Phpactor is requires at least PHP 7.0. If you use a different version of PHP
```
Comment (greg0ire) [inline doc/vim-plugin.md]: requires (no is)

File: doc/vim-plugin.md [hunk 18]
```diff
@@ -0,0 +1,137 @@
...
+locally, you may need to target PHP 7.0, you can change the PHP binary the
+plugin will use as follows:
+
+```
+let g:phpactorPhpBin = "/usr/bin/local/php6"
```
Comment (greg0ire) [inline doc/vim-plugin.md]: lol

**Comment by dantleech** [comment]
@greg0ire thanks for the review, have fixed all the typos, should have used `:set spell` :)

what do you think about the `History` section? interesting or better as a blog post?

**Comment by greg0ire** [comment]
>what do you think about the History section? 
> interesting 

as fuck, I already linked to it here: https://www.reddit.com/r/vim/comments/7q3mth/using_vim_as_a_php_ide/dsnvvu6/
or better as a blog post?

> better as a blog post?

nah it's great to have the "why" built-in the docs. Could be interesting *also* as a blog post, but it definitely has its place here

**Comment by dantleech** [comment]
Merging this version, but will do some more work on it.

**Pull request merged**